### PR TITLE
Add default azure credential sample for blob, in new location under /samples

### PR DESCRIPTION
**Pull request by KieranBrantnerMagee**
As subject says.
Forked from prior PR so that EH and Blob samples can go in distinctly.

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/storage/azure-storage-blob/samples/blob_samples_authentication.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+class AuthSamples(object):
+    oauth_url = "{}://{}.blob.core.windows.net".format(
+        os.getenv(PROTOCOL),
```
Comment (rakshith91) [inline sdk/storage/azure-storage-blob/samples/blob_samples_authentication.py]: should be "PROTOCOL"

Comment (KieranBrantnerMagee) [reply sdk/storage/azure-storage-blob/samples/blob_samples_authentication.py]: /facepalm/  thank you!

**Comment by KieranBrantnerMagee** [comment]
Closing to re-fork and cherry pick against HEAD given rebase changes on the parent branch.

**Pull request closed without merging**